main.go

```diff
@@ -104,10 +105,29 @@ func (b *bufferJoinLines) Write(p []byte) (n int, err error) {
 
 func singleLinePrint(node ast.Node) string {
 	var buf bufferJoinLines
-	printer.Fprint(&buf, token.NewFileSet(), node)
+	printNode(&buf, token.NewFileSet(), node)
 	return buf.String()
 }
 
+func printNode(w io.Writer, fset *token.FileSet, node ast.Node) {
+	switch node := node.(type) {
```

Comment:
I'd prefer `switch x :=` as otherwise it's not clear that `node` is no longer just `ast.Node`

Reply:
tbh, in type switches I generally prefer to use the same variable name, so it's clear that we're talking about exactly the same value (and there's no way to accidentally use the original). It's still a node, after all, even if we happen to have more information about it.

Reply:
I disagree - you use `node` as a slice, so it's clearly not just an `ast.Node` like any other.

Reply:
To clarify - I do like the point on no accidental uses, but at the same time I don't like the shadowing - using `node` as a slice or as any other specific type seems wrong to me.

Reply:
Done, but I still don't think it's worth inventing a new name for something just because it's been type-narrowed.

Reply:
I would usually agree, but `node[i]` or `node1 + node2` seems like taking that to the extreme. You're not narrowing an interface type to another interface type, or even a struct - you're narrowing to a slice, which means it can be used in a very different way.

This is a style nit in any case, so I don't strongly feel either way - but I do appreciate you sticking to the style present elsewhere :)

Reply:
Consistency is indeed a good argument. I guess I always do `switch x := x.(type)` so I don't have to think of a new name (is `x` really any better than `node`?), but I'm happen to go along with the rest of the code.